### PR TITLE
Twilio Verify Readme -- Pre-requisites(3)

### PR DESCRIPTION
**Pull request by camckin10**
'Pre-requisites' of Twilio Verify Devise (Part 3)
Issue https://github.com/MidwestAccessCoalition/twilio-verify-devise/issues/83

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
 
 ## Pre-requisites
 
-To use the Authy API you will need a Twilio Account, [sign up for a free Twilio account here](https://www.twilio.com/try-twilio).
 
-Create an [Authy Application in the Twilio console](https://www.twilio.com/console/authy/applications) and take note of the API key.
+To use the Twilio Verify API you will need a Twilio Account, [sign up for a free Twilio account here](https://www.twilio.com/try-twilio).
+
+Create an [Twilio Verify Application in the Twilio console](https://www.twilio.com/console/authy/applications) and take note of the API key.
```
Comment (wwahammy) [inline README.md:46]: This isn't the Twilio Verify console URL; this is the one for creating an Authy application which doesn't exist anymore.

File: README.md [hunk 2]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+
+## Getting started
+
+First get your Twilio Verify API key from [the Twilio console](https://www.twilio.com/console/authy/applications). We recommend you store your API key as an environment variable.
```
Comment (wwahammy) [inline README.md:50]: Not the right URL

File: README.md [hunk 3]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+$ export TWILIO_AUTH_TOKEN=YOUR_TWILIO_AUTH_TOKEN
+$ export TWILIO_ACCOUNT_SID=YOUR_TWILIO_ACCOUNT_SID
+$ export TWILIO_VERIFY_SERVICE_SID=YOUR_TWILIO_VERIFY_SERVICE_SID
+```
+
+Next add the gem to your Gemfile:
+
+```ruby
+gem 'devise'
+gem 'devise-twilio-verify'
```
Comment (wwahammy) [inline README.md:62]: Please correct the name of the gem. Additionally, we haven't uploaded to rubygems.org, so we need to use the git parameter for the gem requirement. I recommend the following:
 ```suggestion
gem 'devise-authy', git: 'https://github.com/MidwestAccessCoalition/twilio-verify-devise.git'
```

File: README.md [hunk 4]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+```
+
+And then run `bundle install`
+
+Add `Devise Twilio Verify` to your App:
+
+    rails g devise_twilio_verify:install
```
Comment (wwahammy) [inline README.md:69]: We haven't renamed the generator (https://github.com/MidwestAccessCoalition/twilio-verify-devise/blob/main/lib/generators/devise_authy/install_generator.rb). It's still called devise_authy:install

File: README.md [hunk 5]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+### Configuring Models
+
+You can add devise_twilio_verify to your user model in two ways.
+
+#### With the generator
+
+Run the following command:
+
+```bash
+rails g devise_twilio_verify [MODEL_NAME]
```
Comment (wwahammy) [inline README.md:83]: We didn't rename the generator, it's still called devise_authy https://github.com/MidwestAccessCoalition/twilio-verify-devise/blob/main/lib/generators/devise_authy/devise_authy_generator.rb

File: README.md [hunk 6]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+```
+
+To support account locking (recommended), you must add `:twilio_verify_lockable` to the `devise :twilio_verify_authenticatable, ...` configuration in your model as this is not yet supported by the generator.
```
Comment (wwahammy) [inline README.md:86]: We didn't rename these to "twilio_verify_lockable" and "twilio_verify_authenticatable" yet.

File: README.md [hunk 7]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+
+    add_index :users, :authy_id
+  end
+
+  def self.down
+    change_table :users do |t|
+      t.remove :authy_id, :last_sign_in_with_twilio_verify, :twilio_verify_enabled
+    end
+  end
+end
```
Comment (wwahammy) [inline README.md:117]: I don't really know where this came from. It's not accurate, we're still using the authy_id and if the field was removed by the user, it wouldn't work.

File: README.md [hunk 8]
```diff
@@ -40,9 +40,108 @@ This is a [Devise](https://github.com/plataformatec/devise) extension to add [Tw
...
+rake db:migrate
+```
+
+**[Optional]** Update the default routes to point to something like:
+
+```ruby
+devise_for :users, :path_names => {
+	:verify_twilio_verify => "/verify-token",
+	:enable_twilio_verify => "/enable-two-factor",
+	:verify_twilio_verify_installation => "/verify-installation"
```
Comment (wwahammy) [inline README.md:134]: There are no paths matching these. We haven't renamed the paths.

**Review by wwahammy — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (wwahammy) [reply README.md:103]: Again, there are no path names called :verify_twilio_verify, :enable_twilio_verify, etc.

**Review by wwahammy — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -194,15 +184,6 @@ For either method above, run the migrations:
...
 
-**[Optional]** Update the default routes to point to something like:
-
-```ruby
-devise_for :users, :path_names => {
-	:verify_authy => "/verify-token",
-	:enable_authy => "/enable-two-factor",
-	:verify_authy_installation => "/verify-installation"
-}
-```
```
Comment (wwahammy) [inline README.md:137]: This is still accurate and should be kept.

File: README.md [hunk 10]
```diff
@@ -94,16 +94,6 @@ For either method above, run the migrations:
...
-**[Optional]** Update the default routes to point to something like:
-
-```ruby
-devise_for :users, :path_names => {
-	:verify_twilio_verify => "/verify-token",
-	:enable_twilio_verify => "/enable-two-factor",
-	:verify_twilio_verify_installation => "/verify-installation"
-}
-```
-
```
Comment (wwahammy) [inline README.md]: Yes, please delete this.

**Review by wwahammy — CHANGES_REQUESTED** [review]

**Review by BintLopez — APPROVED** [review]
🦑

**Review by wwahammy — APPROVED** [review]

**Pull request merged**